pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp

```diff
@@ -692,6 +692,142 @@ class StateVectorCudaManaged
         }
         return expval(obsName, wires, params, matrix_cu);
     }
+    /**
+     * @brief See `expval(std::vector<CFP_t> &gate_matrix = {})`
+     */
+    auto expval(const std::vector<size_t> &wires,
+                const std::vector<std::complex<Precision>> &gate_matrix) {
+
+        std::vector<CFP_t> matrix_cu(gate_matrix.size());
+
+        for (std::size_t i = 0; i < gate_matrix.size(); i++) {
+            matrix_cu[i] =
+                cuUtil::complexToCu<std::complex<Precision>>(gate_matrix[i]);
+        }
+
+        if (gate_matrix.empty()) {
+            std::string message = "Currently unsupported observable";
+            throw LightningException(message.c_str());
+        }
+
+        auto &&local_wires =
+            (gate_matrix.empty())
+                ? wires
+                : std::vector<size_t>{wires.rbegin(), wires.rend()};
+
+        auto expect_val =
+            getExpectationValueDeviceMatrix(matrix_cu.data(), local_wires);
+        return expect_val;
+    }
+
+    /**
+     * @brief Get expectation of a given host or device defined array.
+     *
+     * @param matrix Host or device defined row-major order gate matrix array.
+     * @param tgts Target qubits.
+     * @return auto Expectation value.
```

Comment:
Can you update the docstring here to include the template param as `@tparam` and the remaining arguments that are missing for the sparse format.